internal/builder/builder.go

```diff
@@ -0,0 +1,42 @@
+// Package builder has function, which
+// build necessary shape.
+package builder
+
+import (
+	"github.com/shopspring/decimal"
+	"volume_calculator/internal/shapes"
+)
+
+type Builder struct {
+	shape                         string
+	radius, length, width, hieght decimal.Decimal
+}
+
+func NewBuilder(shape string, radius, length, width, hieght decimal.Decimal) *Builder {
+	return &Builder{shape: shape, radius: radius, length: length, width: width, hieght: hieght}
+}
+
+func (builder Builder) BuildShape() shapes.Volumer{
+	var sh shapes.Volumer
+	switch builder.shape {
+	case "sphere":
+		sh = builder.buildSphere(builder.radius)
+	case "pyramid":
+		sh = builder.buildPyramid(builder.length, builder.width, builder.hieght)
+	case "cylinder":
+		sh = builder.buildCylinder(builder.hieght, builder.radius)
+	}
+	return sh
```

Comment:
in case on unknown shape you'll get panic with null pointer dereference error. Need to handle this case